server/src/main/java/io/crate/analyze/relations/UnionSelect.java

```diff
@@ -53,17 +55,7 @@ public UnionSelect(AnalyzedRelation left, AnalyzedRelation right, boolean isDist
         this.right = right;
         this.name = new RelationName(null, UUIDs.dirtyUUID().toString());
         // SQL semantics dictate that UNION uses the column names from the first relation (top or left side)
-        List<Symbol> fieldsFromLeft = left.outputs();
-        ArrayList<ScopedSymbol> outputs = new ArrayList<>(fieldsFromLeft.size());
-        for (int i = 0; i < fieldsFromLeft.size(); i++) {
-            Symbol field = fieldsFromLeft.get(i);
-            Symbol rightField = right.outputs().get(i);
-            DataType<?> type = field.valueType().precedes(rightField.valueType())
-                ? field.valueType()
-                : rightField.valueType();
-            outputs.add(new ScopedSymbol(name, Symbols.pathFromSymbol(field), type));
-        }
-        this.outputs = List.copyOf(outputs);
+        this.outputs = List.copyOf(unionizeOutputs(left.outputs(), right.outputs()));
```

Comment:
Why do we need this? Bias for the left relation for union should be fine. No?

Even if not, this seems to be unrelated to the main change of this PR, which afaik is to address this:

```
cr> select o['unknown'] from t;                                                                                                                   
ColumnUnknownException[Column o['unknown'] unknown]

cr> select o['unknown'] from t as t2; -- should throw like above            
```

Reply:
Could it be because of a case where `o['uknown']` exists in left relation but not in right?

Reply:
I am adding a step to validate the object key exists in the `ObjectType` before allocating subscript functions which breaks analysis of `union` if selecting sub-column of object from right table.

```
create table v1 (obj object as (a int));
create table v2 (obj object as (b int));
select obj['a'], obj['b'] from (select obj from v1 union all select obj from v2) v; -- cannot select 'b' without merging 'obj'
```